README.md

```diff
@@ -144,29 +148,57 @@ If you project is compatible with the `go` tool, you can generate and update
 your `BUILD` files automatically using [Gazelle](go/tools/gazelle/README.md),
 a command line tool which is part of this repository.
 
-* You can install Gazelle using the command below. This assumes this repository
+* The gazelle rule in your root build file gives you the ability to build and 
+  run gazelle on your project, this is the preferred way to use it. If you want 
+  to you can also install Gazelle using the command below. This assumes this 
+  repository
   is checked out under [GOPATH](https://github.com/golang/go/wiki/GOPATH).
 
 ```
 go install github.com/bazelbuild/rules_go/go/tools/gazelle/gazelle
 ```
 
-* To run Gazelle for the first time, run the command below from your project
-  root directory.
+* To run Gazelle and update your `BUILD` files, run the command below from any 
+  diretory in your project.
 
 ```
-gazelle -go_prefix github.com/joe/project
+bazel run //:gazelle
 ```
 
-* To update your `BUILD` files later, just run `gazelle`.
 * By default, Gazelle assumes external dependencies are present in
   your `WORKSPACE` file, following a certain naming convention. For example, it
   expects the repository for `github.com/jane/utils` to be named
-  `@com_github_jane_utils`. If you prefer to use vendoring, run `gazelle` with
-  `-external vendored`. See [Vendoring.md](Vendoring.md).
+  `@com_github_jane_utils`. If you prefer to use vendoring, add `external=vendored` 
+  to the gazelle rule. See [Vendoring.md](Vendoring.md).
 
 See the [Gazelle README](go/tools/gazelle/README.md) for more information.
 
+## Build modes
+
+### Building static binaries
+
+You can build binaries in static linking mode using
+```
+bazel build --output_groups=static //:my_binary
+```
+
+### Using the race detector
+
+You can run tests with the race detector enabled using
+```
+bazel test --features=race //...
+```
+
+You can build binaries with the race detector enabled using
+```
+bazel test --output_groups=race //...
```

Comment:
I think so, yes. I'm fixing this and rewording the paragraph below in #666. Hopefully it's a little clearer.